packages/extension-ui/src/components/SplashHandler.tsx

```diff
@@ -40,10 +41,36 @@ function SplashHandler({ children, className }: SplashHandlerProps): React.React
     exiting: { opacity: 0 }
   };
 
+  useEffect(() => {
+    if (!isContentVisible) {
+      return;
+    }
+
+    const updateWithErrorLog = (prevIsSplashOn: boolean) => {
+      if (prevIsSplashOn) {
+        console.error('Fallback timeout needed to turn off splash video.');
+      }
+
+      return false;
+    };
+
+    const timeoutId = setTimeout(setIsSplashOn, 2000, updateWithErrorLog);
```

Comment:
Why is that actually needed?

Reply:
it's not, I just thought, that there might be (I assume) one thousand reasons for video to not call `onEnd`, so while we should handle them I think this fallback is nice to have